hypershift-operator/main.go

```diff
@@ -213,6 +215,40 @@ func run(ctx context.Context, opts *StartOptions, log logr.Logger) error {
 	otel.SetTracerProvider(tp)
 	otel.SetTextMapPropagator(propagation.NewCompositeTextMapPropagator(propagation.Baggage{}, propagation.TraceContext{}))
 
+	// Block default ingress controller from admitting HCP private routes
+	ic := &operatorv1.IngressController{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "default",
+			Namespace: "openshift-ingress-operator",
+		},
+	}
+	if _, err := controllerutil.CreateOrUpdate(ctx, mgr.GetClient(), ic, func() error {
+		if ic.Spec.Replicas == nil {
+			return fmt.Errorf("default ingress controller not found")
+		}
+		if ic.Spec.RouteSelector == nil {
+			ic.Spec.RouteSelector = &metav1.LabelSelector{}
+		}
+		if ic.Spec.RouteSelector.MatchExpressions == nil {
+			ic.Spec.RouteSelector.MatchExpressions = []metav1.LabelSelectorRequirement{}
+		}
+		for _, requirement := range ic.Spec.RouteSelector.MatchExpressions {
+			if requirement.Key != "hypershift.openshift.io/hosted-control-plane" {
```

Comment:
can put `hypershift.openshift.io/hosted-control-plane` into a constant since it's used twice